platform-hub-api/spec/jobs/tokens_sync_job_spec.rb

```diff
@@ -61,6 +61,8 @@
         expect(KubernetesCluster).to receive(:names).never
         expect(Kubernetes::TokenSyncService).to receive(:sync_tokens).never
         expect(AuditService).to receive(:log).never
+
+        TokensSyncJob.new.perform
```

Comment:
Won't you need to call this before the assertions?

Reply:
Not neccessarily – check out https://relishapp.com/rspec/rspec-mocks/v/3-8/docs/basics/expecting-messages.

There is a way of making expectation assertions _after_ the operation, using [spies](https://relishapp.com/rspec/rspec-mocks/v/3-8/docs/basics/spies), but I tend to prefer setting assertions up beforehand (no particular reason apart from style and habit, really).